python/lsst/ip/diffim/dcrModel.py

```diff
@@ -83,22 +82,23 @@ def fromImage(cls, maskedImage, dcrNumSubfilters, filterInfo=None, psf=None):
         """
         # NANs will potentially contaminate the entire image,
         # depending on the shift or convolution type used.
-        model = maskedImage.clone()
-        badPixels = np.isnan(model.image.array) | np.isnan(model.variance.array)
-        model.image.array[badPixels] = 0.
-        model.variance.array[badPixels] = 0.
-        model.image.array /= dcrNumSubfilters
+        model = maskedImage.image.clone()
+        mask = maskedImage.mask.clone()
         # We divide the variance by N and not N**2 because we will assume each
         # subfilter is independent. That means that the significance of
         # detected sources will be lower by a factor of sqrt(N) in the
         # subfilter images, but we will recover it when we combine the
         # subfilter images to construct matched templates.
-        model.variance.array /= dcrNumSubfilters
-        model.mask.array[badPixels] = model.mask.getPlaneBitMask("NO_DATA")
+        variance = maskedImage.variance.clone()
+        variance /= dcrNumSubfilters
+        badPixels = np.isnan(model.array) | np.isnan(variance.array)
```

Comment:
Shouldn't these pixels be flagged by the pipeline?

Reply:
They should be, but I am paranoid since everything will fail if there is a single NAN pixel that slips through. I could add a warning for if that ever happens.

Reply:
Based on Robert's comment, I'll replace this with raising a ValueError should there ever somehow be a NAN value in the maskedImage.

Reply:
In the unmasked part of the image, no?

Reply:
Yes, of course, and that reminds me why I originally just set any NAN pixel to zero. For simplicity, I perform some operations on the entire array, and that throws `RuntimeWarning`s if there are any NAN values. Any objection if I simply set "NO_DATA" pixels to zero, then raise a `ValueError` if there are any remaining NANs?

Reply:
Where does the RuntimeWarning come from? We should be able to process data with NaNs in pixels that we don't actually care about (i.e. masked).

How does it change the DCR results if a masked pixel is set to `0` or some other random value?

Reply:
We interpolate over all NaNs, as otherwise e.g. smoothing kernels spread out the damage.  The stack consistently assumes that bad things like NaNs and dropouts are interpolated with a bit set.

Reply:
Actually I was wrong about that warning. The RuntimeWarning was caused by `dcrWeights` being very small, which was fixed in the commit earlier this afternoon. It looks as though everything would actually work without the NAN test, since it does get masked later. So, I could remove these lines if you would prefer.

Reply:
Yes, lets try removing this. If we do end up with NaN-related failures later on, we can try to do a better fix.